docs/current_docs/manuals/developer/functions.mdx

```diff
@@ -213,32 +329,38 @@ The result will look like this:
 Hello, world
 ```
 
-:::tip
-For consistency between Python and the API, choose to always set optionals
-with a default value, even if it's:
-
-```python
-name: str | None = None
-```
-The above is in fact necessary if the default value needs to be set from the
-function body instead of the signature (dynamic, complex or a mutable value).
-:::
```

Comment:
The reason for this is that if I started with `name: str | None = None` it has both concepts of an optional and a default value. So to introduce these concepts in turn, I had to put optionals first but it led to an unnatural function signature for Python.

As I explained in my last comment, I'm ok removing this and fixing it in a later PR in relation to:
- https://github.com/dagger/dagger/issues/6749